### ISSUE TITLE
Travis deploy cloud foundry Incorrect Usage: unknown flag`u'

### ISSUE THREAD

**Issue by aemrie** [travis-ci/dpl#512]
```
deploy:
  edge: true
  provider: cloudfoundry
  api: https://api.ng.bluemix.net
  username: ****
  password:
    secure: *********
  organization: ******
  space: *****
  on:
    branch: master
    repo: *****
```

> After correctly setting the API endpoint, Travis CI log states...
```
API endpoint: https://api.ng.bluemix.net (API version: 2.54.0)
Not logged in. Use 'cf login' to log in.
Incorrect Usage: unknown flag`u'
```
Username is simple email address that does not contain any special characters (Eg. user@***.com)

**Comment by BanzaiMan** [comment]
Looks like flags to `./cf` are wrong. Could you try this?

```yaml
deploy:
  edge:
    branch: cf-login # instead of true
    ⋮ # rest
```

**Comment by textbook** [comment]
That fixes it for my project (compare builds [50](https://travis-ci.org/textbook/known-for-web/builds/162438249) and [51](https://travis-ci.org/textbook/known-for-web/builds/162439946)) and is consistent with the [CF CLI docs](http://cli.cloudfoundry.org/en-US/cf/login.html).

**Comment by aemrie** [comment]
> Updated new travis config per suggestion

```
deploy:
  edge: 
     branch: cf-login
  provider: cloudfoundry
  api: https://api.ng.bluemix.net
  username: ****
  password:
    secure: *********
  organization: ******
  space: *****
  on:
    branch: master
    repo: *****

```

> Now getting slightly different error in Travis logs for gem 'dpl-*.gem'.

```
Building dpl gem locally with source travis-ci/dpl and branch cf-login
0.00s
0.44s$ git clone https://github.com/travis-ci/dpl travis-ci/dpl
Cloning into 'travis-ci/dpl'...
remote: Counting objects: 6740, done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 6740 (delta 3), reused 0 (delta 0), pack-reused 6736
Receiving objects: 100% (6740/6740), 1.94 MiB | 0 bytes/s, done.
Resolving deltas: 100% (4057/4057), done.
Checking connectivity... done.
0.00s
0.02s$ git checkout cf-login
Branch cf-login set up to track remote branch cf-login from origin.
Switched to a new branch 'cf-login'
0.37s$ rvm 1.9.3 --fuzzy do ruby -S gem build dpl.gemspec
WARNING:  pessimistic dependency on rspec (~> 3.0.0, development) may be overly strict
  if rspec is semantically versioned, use:
    add_development_dependency 'rspec', '~> 3.0', '>= 3.0.0'
WARNING:  open-ended dependency on rspec-its (>= 0, development) is not recommended
  if rspec-its is semantically versioned, use:
    add_development_dependency 'rspec-its', '~> 0'
WARNING:  open-ended dependency on rake (>= 0, development) is not recommended
  if rake is semantically versioned, use:
    add_development_dependency 'rake', '~> 0'
WARNING:  open-ended dependency on coveralls (>= 0, development) is not recommended
  if coveralls is semantically versioned, use:
    add_development_dependency 'coveralls', '~> 0'
WARNING:  See http://guides.rubygems.org/specification-reference/ for help
  Successfully built RubyGem
  Name: dpl
  Version: 1.8.20.travis.20.1
  File: dpl-1.8.20.travis.20.1.gem
0.00s
0.00s
0.01s
0.00s
0.00s
0.40sERROR:  Could not find a valid gem 'dpl-*.gem' (>= 0) in any repository
The command "rvm 1.9.3 --fuzzy do ruby -S gem install dpl-*.gem --local --pre" failed and exited with 2 during .
```

**Comment by BanzaiMan** [comment]
@aemrie Do you have a full build log for this failure?

**Comment by BanzaiMan** [comment]
Fixed with 1.8.20.
